dds/DCPS/RTPS/RtpsDiscovery.cpp

```diff
@@ -712,14 +712,14 @@ RtpsDiscovery::Config::discovery_config(ACE_Configuration_Heap& cf)
         } else if (name == "UseXTypes") {
           const OPENDDS_STRING& value = it->second;
           int smInt;
-          if (!DCPS::convertToInteger(value, smInt)) {
+          if (!DCPS::convertToInteger(value, smInt) || smInt < 0 || smInt > 2) {
```

Comment:
For configuration parsing, how about allowing 0 or 1 (for backwards compatibility) or a string value for the enumerator?

Reply:
This allows `UseXTypes` to be 0, 1, 2 for "non-XTypes", "Minimal", and "Complete" respectively. What else am I missing?

Reply:
I meant allowing a config file that has the string in it, `UseXTypes=Complete`